demos/common/python/models/hpe_associative_embedding.py

```diff
@@ -81,19 +84,33 @@ def preprocess(self, inputs):
             self.logger.warn("Chosen model aspect ratio doesn't match image aspect ratio")
         resize_img_scale = np.array((inputs.shape[1] / w, inputs.shape[0] / h), np.float32)
 
-        img = np.pad(img, ((0, self.h - h), (0, self.w - w), (0, 0)),
+        if self.padding_mode == 'center':
+            pad = [(self.h - h + 1) // 2, (self.w - w + 1) // 2, (self.h - h) // 2, (self.w - w) // 2]
+        else:
+            pad = [0, 0, self.h - h, self.w - w]
+        img = np.pad(img, ((pad[0], pad[2]), (pad[1], pad[3]), (0, 0)),
```

Comment:
You should initialize `pad` in correct order. So there won't be that permutation.
BTW `pad` should be a tuple.

Reply:
Changed